crypto/modes/gcm128.c

```diff
@@ -537,7 +537,7 @@ static void gcm_ghash_4bit(u64 Xi[2], const u128 Htable[16],
             Xi[0] = Z.hi;
             Xi[1] = Z.lo;
         }
-    } while (inp += 16, len -= 16);
+    } while (inp += 16, (len -= 16)>0);
```

Comment:
but len is size_t so this is just more ugly IMHO

Reply:
I agree, I changed the condition but I am afraid it is still more ugly than a simple comment that `len % 16 == 0`? What do you think?

Reply:
Alternatively we could ossl_assert that len % 16 == 0.